docs_src/design/ucr/0002-Device-Parent-Child-Relationships.md

```diff
@@ -0,0 +1,109 @@
+
+Since there are multiple similar uses for this relationship information on the north side, it is proposed to locate
+this relationship metadata in the Device object as accessed from core-metadata by all services, rather than to 
+locate it in each north-bound service (which would be particularly problematic for the UI, which gets its data through REST APIs).
+
+The sound-bound Device Service that creates a Device is ideally the service which establishes this relationship data, though it is possible that it is unaware of the parent-child relationship. It should be permitted, therefore, for this relationship information to also be set by north-bound services (most likely the UI) and simply ignored by the south-bound Device Service.
+
+A potential solution might add a field to the device structure such as
+```json
+"relationships": [
+        "parent": "First-Floor-Gateway"
+      ]
+```
+It is probably also necessary to indicate which device is the "main" or "publisher" device (ie, the gateway device), 
+as any devices without a configured relationship will probably be inferred to be children of that device.
+
+### Extensions to the main Use Case
+1. In addition to parent-child relationships, other relationships might be indicated in this metadata, such as an
+"extends" when an analytic service extends an existing south-bound device, adding new Device Resources beyond what the south-bound service provides.
+2. Some services add "devices" which have no physical counterpart, eg, for an NTP client service where the "device" 
+serves simply as a container for the Resources necessary to configure and report the status of the service.
+In these cases, it would be helpful for the other services if it described itself as something like a "system" device,
+meaning one that doesn't have a physical (south-bound or hardware) counterpart.
+3. It could be possible for north-bound or analytic services to add metadata of their choice, related to an individual device, in this "relationships" area. 
+This definitely muddies the clear objective of this UCR, but consideration of it might steer the ultimate solution. 
+As application service developers, we have felt at times the desire to "annotate" a device with something more formal than just another label; 
+if other reviewers agree with this notion, it can be adopted here, since it follows the same lines 
+(metadata that other services might add to a device).
+
+
+### Existing solutions
+The Device structure in Eaton's legacy products indicated this parent-child relationship bidirectionally: each device indicated its parent device (if any) with one field, and its child devices (if any) with a list of IDs.
```

Comment:
bi-directional relationships are a bit harder to maintain.  Given the requirements of north bound systems above, can you shed some light on how/why they would need the bidirectional relationship?

Reply:
This one I had to think about for a long time, which is why it took a while to post revisions. I had been thinking of just a forward (one-way) parent=x.y.z approach, but realized that we do want a means to learn the children. So I added something about that as a requirement, without specifying.